plugins/woocommerce/src/StoreApi/Formatters/MoneyFormatter.php

```diff
@@ -23,12 +24,30 @@ public function format( $value, array $options = [] ) {
 			]
 		);
 
-		return (string) intval(
-			round(
-				( (float) wc_format_decimal( $value ) ) * ( 10 ** absint( $options['decimals'] ) ),
-				0,
-				absint( $options['rounding_mode'] )
-			)
-		);
+		// Ensure rounding mode is valid.
+		$rounding_modes           = [ PHP_ROUND_HALF_UP, PHP_ROUND_HALF_DOWN, PHP_ROUND_HALF_EVEN, PHP_ROUND_HALF_ODD ];
+		$options['rounding_mode'] = absint( $options['rounding_mode'] );
+		if ( ! in_array( $options['rounding_mode'], $rounding_modes, true ) ) {
+			$options['rounding_mode'] = PHP_ROUND_HALF_UP;
+		}
+
+		if ( ! is_int( $value ) && ! is_string( $value ) && ! is_float( $value ) ) {
+			wc_doing_it_wrong(
+				__FUNCTION__,
+				__( 'This function expects a $value arg of type INT, STRING or FLOAT.', 'woocommerce' ),
```

Comment:
I'm not sure we should be translating this (though, yes, I know there are many existing places in the code base where we *do* translate logging messages).

I'm thinking here of our [logging best practices](https://github.com/woocommerce/woocommerce/blob/9.1.1/docs/extension-development/logging.md#best-practices) (though mostly written in relation to `wc_get_logger()`, I think many of the principles carry over to logging in general).

Reply:
Agree with this, especially since developers would likely search the error online, having different errors per language reduce the chances of them finding a relevant discussion/fix.

Reply:
I added it because, indeed, it appears often throughout our codebase.

removed in https://github.com/woocommerce/woocommerce/pull/49361/commits/3767dedfcb114689066e4ac5107ddb806efa11e0